### PR TITLE
Update docs with v2 compatible examples

### LINKED ISSUE #5853: [v2] Update docs to cover breaking changes in v2
## Summary

Now that the v2 API's have mostly settled down, make a pass through Gatsby's docs and update them to reference the correct parts of Gatsby's API for v2. 

Remove references to layouts, add mentions of StaticQuery. Document `pageContext` (previously `pathContext`, but it doesn't seem like there's any docs for that).

### Basic example

The [_Querying with GraphQL_](https://v2--gatsbyjs.netlify.com/docs/querying-with-graphql/) doc:

- doesn't mention StaticQuery, which it should
- does mention Layouts, which it shouldn't

### Motivation

Docs should be in sync with API updates.

### PR DESCRIPTION
**Pull request by kkemple**
Took a last pass through docs and READMEs looking for examples and docs that make references to v1 APIs and updating them to v2 APIs

Closes #5853

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview for *using-glamor* failed.

Built with commit 7ff4f994cbd1b02ed9065b74edb58be3b0b24758

https://app.netlify.com/sites/using-glamor/deploys/5b7b1b81c6aed62c1f8ed5fb

**Comment by KyleAMathews** [comment]
Deploy preview for *image-processing* failed.

Built with commit 7ff4f994cbd1b02ed9065b74edb58be3b0b24758

https://app.netlify.com/sites/image-processing/deploys/5b7b1b81c6aed62c1f8ed5f8

**Comment by KyleAMathews** [comment]
Deploy preview for *using-contentful* failed.

Built with commit 7ff4f994cbd1b02ed9065b74edb58be3b0b24758

https://app.netlify.com/sites/using-contentful/deploys/5b7b1b87c6aed62c1f8ed601

File: packages/gatsby/src/redux/actions.js [hunk 1]
```diff
@@ -1134,11 +1134,11 @@ actions.addThirdPartySchema = (
...
 
 /**
- * All defined actions.
+ * All action creators wrapped with a dispatch.
  */
 exports.actions = actions
 
 /**
- * All action creators wrapped with a dispatch.
+ * All action creators wrapped with a dispatch. - *DEPRECATED*
```
Comment (jlengstorf) [inline packages/gatsby/src/redux/actions.js]: Are we doing JSDoc in this repo? If so, I think a `@deprecated` tag would help here.

**Pull request merged**